ankidroid/Anki-Android-Backend
# 2.1.66 and the removal of legacy schema support

2.1.66 is not due out soon, but I thought I'd do this now while the proto gen changes are fresh in memory.

- Update the backend protobuf generation to match the refactoring done
on the desktop code recently. This involved switching from the Python-based
script to a new one in Rust. Comments from the proto files are now
included in the generated backend bindings.

Also:

- Rumo contributed changes to the upstream Chrono crate which make
it work properly on Android, and these changes have been merged in,
so AnkiDroid no longer needs to pin an older version.
- Update to latest NDK, as latest Rust requires it (closes #180 and #179).
- Update to latest Rust
- Update to Gradle 8, closes #248, #245, #230, #171
- Add a workaround for a crashing bug in the sim caused by the above.
Needs testing to make sure building/running on macOS still works.
- Switch from protobuf to protobuf-lite, which is what Google recommend
for use on Android.
- Add a script to make it easier to reformat Rust code when it fails
checks.
- Plus more changes in the later commits
